### PR TITLE
Issue-140 Implementation login servises and handlers

### PR DESCRIPTION
**Pull request by PavloRymaruck**

### CODE REVIEW

File: pkg/ims/service/service.go [hunk 1]
```diff
@@ -140,4 +144,81 @@ func (service *SignUpService) EmailVerificationToken (token *models.EmailVerific
...
+		return err
+	}
+	if user.Email == "" {
+		return err
+	}
+
+	id = strconv.Itoa(user.ID)
+	cred, err = credRepo.GetCredentialByUserID(id)
+	pwd := []byte(request.Password + cred.Salt)
+	hashed_pasword := []byte(cred.PasswordHash)
```
Comment (taaraora) [inline pkg/ims/service/service.go:167]: please use camelCase instead of snake_case

File: pkg/ims/service/service.go [hunk 2]
```diff
@@ -140,4 +144,81 @@ func (service *SignUpService) EmailVerificationToken (token *models.EmailVerific
...
+	err = bcrypt.CompareHashAndPassword(hashed_pasword, pwd)
+	if err != nil {
+		return err
+	}
+
+	mySigningKey := []byte("AllYourBase")
```
Comment (taaraora) [inline pkg/ims/service/service.go:173]: it should be injected in service while creation.

File: pkg/ims/validation/data.go [hunk 3]
```diff
@@ -43,4 +43,31 @@ func Data (request *models.SignupRequest) error {
 	}
 
 	return nil
-}
\ No newline at end of file
+}
+
+func Data_login(request *models.LoginRequest) error {
```
Comment (taaraora) [inline pkg/ims/validation/data.go:48]: do not use snake case

**Comment by PavloRymaruck** [comment]
closes #140

**Pull request merged**